### ISSUE TITLE
twisted.conch.test.test_conch tests leave the reactor unclean

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2267]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#2267|
|Type|defect|
|Created|2006-12-06 02:49:42Z|

They are also terrible

<details><summary>Searchable metadata</summary>

```
trac-id__2267 2267
type__defect defect
reporter__jml jml
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__tests tests
time__1165373382000000 1165373382000000
changetime__1168753080000000 1168753080000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_1"></a>|@radix commented|
|-|-|

I've started a branch named conch-cleanliness-2267 with the conch changes from unclean-errors-2091-5.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_4"></a>|@jerub set owner to @radix|
|-|-|

There are several _getDeferred() functions defined, but the Deferred() objects are created elsewhere and set on the objects. It would be nicer if they were at least initialised to be None or Deferred(), instead of arbitarily set upon the object at some point later (such as OpenSSHClientTestCase.execute).

execute methods all don't have docstrings. I would like it if they did. At least each test class should have a docstring on either its execute method or on the class proper, to indicate what the intent is.

UnixClientTestCase.execute has an abundance of lambdas. I suggest that this:
```python
        d.addCallback(lambda x : p.deferred)
        def cleanup(buf):
            d = defer.maybeDeferred(conn.transport.transport.loseConnection)
            d.addCallback(lambda x: buf)
            return d
        d.addCallback(cleanup)
```
be factored differently, especially as in this case 'buf' is actually a Deferred instance....

Thanks for doing this cleanup, unfortunately there's more cleaning to do, but we're getting there. :)

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_5"></a>|@jerub set owner to @jml|
|-|-|

Merge this NOW NOW  NOW WNOWN NOWNOW NOWNOWN

please :)

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_6"></a>|@jml set status to `closed`|
|-|-|

(In [19320]) Make sure that certain conch tests do not tarnish the reactor state.
 * Author: jml, radix
 * Reviewer: jerub
 * Fixes [#2267](https://github.com/twisted/twisted/issues/2267)
 * Refs [#2091](https://github.com/twisted/twisted/issues/2091)

Clean up a lot of cruft in test_cftp and test_conch. The tests are still
pretty bad, but at least their badness is clear.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
